### PR TITLE
cli/start: make SIGTERM idempotent

### LINKED ISSUE #48070: cli/start: don't do hard shutdown on 2nd SIGTERM
Today we handle double SIGINT (Ctrl+C) and double SIGTERM the same way: the first initiates a soft shutdown, the second one initiates a hard shutdown.

For SIGINT, the UX is justified: the user is calling `start` interactively, and pressing Ctrl+C twice means "I know what I'm doing, go away faster".

Not so with SIGTERM. This is the signal sent by a process manager. Under normal operation, a process manager would issue
- SIGTERM first,
- wait for a bit
- issue SIGKILL if the process did not shut down on their own.

There is no need to give special meaning to a 2nd SIGTERM to be a hard shutdown. In fact, it's counter productive: if two operators concurrently (perhaps mistakenly) issue a shutdown order to their orchestration tool, we don't want the 2nd order to transform the 1st one in a hard shutdown which the 1st operator did not expect.

So the code should be changed to only handle a double SIGINT as hard shutdown, and ignore a 2nd SIGTERM (and perhaps SIGQUIT) once the graceful shutdown has been initiated.

### PR DESCRIPTION
**Pull request by knz**
First commit from #50538.
Fixes #48070.

Prior to this patch, if SIGTERM was received a 2nd time, the server
would expedite shutdown.

This is inconvenient in practice because if an operator mistakenly
issues a shutdown without being aware that a shutdown is already in
progress, they will get a hard shutdown while expecting a graceful
shutdown. This can incur undesired downtime.

Release note (backward-incompatible change): A node will not any
more proceed to a hard shutdown upon receiving SIGTERM a second time,
or upon receiving SIGTERM after another signal. Instead, the first
SIGTERM initiates a graceful shutdown and further occurrences of
SIGTERM are ignored. To initiate a hard shutdown, the user can issue
SIGINT two times (or SIGINT once after SIGTERM) instead.

### CODE REVIEW

File: pkg/cli/interactive_tests/test_server_sig.tcl [hunk 1]
```diff
@@ -86,7 +86,40 @@ set spawn_id $client_spawn_id
...
+set spawn_id $shell_spawn_id
+system "kill `cat server_pid`"
+eexpect "graceful shutdown"
+system "kill `cat server_pid`"
+# There's still a very small chance the server could finish draining
+# before the second signal is sent, but oh well.
+expect {
+    "shutdown completed" {}
+    "hard shutdown" {
+	report "unexpected hard shutdown"
```
Comment (irfansharif) [inline pkg/cli/interactive_tests/test_server_sig.tcl:114]: I think something is off with the indentation here.

**Review by irfansharif — APPROVED** [review]
:lgtm:

File: pkg/cli/interactive_tests/test_server_sig.tcl [hunk 2]
```diff
@@ -86,7 +86,40 @@ set spawn_id $client_spawn_id
...
+	exit 1
+    }
+    timeout { handle_timeout "server shutdown message" }
+}
+eexpect ":/# "
+end_test
+
+# terminate the SQL shell
```
Comment (irfansharif) [inline pkg/cli/interactive_tests/test_server_sig.tcl:122]: [nit] Capitalize here.

**Review by knz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/50539)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @bdarnell and @irfansharif)

___
*[pkg/cli/interactive_tests/test_server_sig.tcl, line 114 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/50539#-MAWItmMqLeimzXjLMD8-r2-114:-MAgfYx12DRkggI40iNh:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b322b8e6687b77965eb6bd242b7c653da6468b49/pkg/cli/interactive_tests/test_server_sig.tcl#L114)):*
<details><summary><i>Previously, irfansharif (irfan sharif) wrote…</i></summary><blockquote>

I think something is off with the indentation here.
</blockquote></details>

Done.
___
*[pkg/cli/interactive_tests/test_server_sig.tcl, line 122 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/50539#-MAWItmMqLeimzXjLMD8-r2-122:-MAgg3VW8Cb9gGIrdVn_:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b322b8e6687b77965eb6bd242b7c653da6468b49/pkg/cli/interactive_tests/test_server_sig.tcl#L122)):*
<details><summary><i>Previously, irfansharif (irfan sharif) wrote…</i></summary><blockquote>

[nit] Capitalize here.
</blockquote></details>

Done.

**Comment by knz** [comment]
thank you!

bors r=irfansharif

**Pull request merged**